pkg/config/default.go

```diff
@@ -188,6 +189,25 @@ const (
 	DefaultVolumePluginTimeout = 5
 )
 
+func defaultSigPath() (string, error) {
+	// NOTE: For now we want Windows to use system locations.
+	// GetRootlessUID == -1 on Windows, so exclude negative range
+	if unshare.GetRootlessUID() > 0 {
+		configHome, err := homedir.GetConfigHome()
+		if err == nil {
+			sigPath := filepath.Join(configHome, DefaultRootlessSignaturePolicyPath)
+			if err := fileutils.Exists(sigPath); err == nil {
+				return sigPath, nil
+			}
+		} else {
+			if !errors.Is(err, fs.ErrNotExist) {
+				return "", err
+			}
+		}
+	}
+	return DefaultSignaturePolicyPath, nil
```

Comment:
There’s, also… a _whole another_ discussion topic in this problem space: In general, c/image callers should not have their own default path logic _at all_ because they are all going to diverge over time.

Compare https://github.com/containers/image/issues/1726 .

All of this code really should not exist in the first place.

… but, a reminder, removing this code would not really be a fix for https://github.com/containers/podman/issues/23818 .

---

… now, c/image does look for a policy.json in the home directory, but it currently uses a hard-coded `.config/containers/policy.json` and it does not interpret $XDG_CONFIG_HOME ( https://github.com/containers/image/issues/2297 ). So just removing this code right now could break things.